Gordon360/Controllers/RecIMController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Gordon360.Authorization;
+using Gordon360.Models.CCT;
+using Gordon360.Models.CCT.Context;
+using Gordon360.Models.ViewModels;
+using Gordon360.Services;
+using Gordon360.Static.Names;
+using Gordon360.Utilities;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Controllers
+{
+    [Route("api/[controller]")]
+    public class RecIMController : GordonControllerBase
+    {
+        private readonly IRecIMService _recimService;
+
+        public RecIMController(IRecIMService recimService)
+        {
+            _recimService = recimService;
+        }
+
+        ///<summary>Gets a list of all Leagues</summary>
+        /// <returns>
+        /// All Existing Leagues 
+        /// </returns>
+        [HttpGet]
+        [Route("league")]
```

Comment:
Is league a good route for all leagues? Or should it have a different route name? Perhaps league/all? Just looking for the philosophy on this.

Reply:
wasn't anything specific, I was following things specified on the API routes.
 I do agree with league/all

Reply:
Traditionally, REST API endpoints are organized around resources, in this case `leagues`, and HTTP verbs. So, for examples, the endpoint to retrieve all leagues would be `GET /leagues`, and then to retrieve a specific league you would call `/leagues/leagueId` (or `/leagues?Id={ID}`).

Beginning resources for REST API design:

- [Microsoft Best Practices](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design)
- [Stack Overflow Blog](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/)

Reply:
I see, I'll make sure to follow that route design
- [Microsoft's Example](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design#define-api-operations-in-terms-of-http-methods)
- 


> Traditionally, REST API endpoints are organized around resources, in this case `leagues`, and HTTP verbs. So, for examples, the endpoint to retrieve all leagues would be `GET /leagues`, and then to retrieve a specific league you would call `/leagues/leagueId` (or `/leagues?Id={ID}`).
> 
> Beginning resources for REST API design:
> 
> * [Microsoft Best Practices](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design)
> * [Stack Overflow Blog](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/)

Got it, example can be found here:
- [microsoft link](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design#define-api-operations-in-terms-of-http-methods)